### PR TITLE
dimension inequality

### PR DESCRIPTION
**Pull request by jonsterling**
~~This is extremely slow and badly written.~~ The 🧞 has returned!

### CODE REVIEW

File: test/circle.cooltt [hunk 1]
```diff
@@ -17,4 +17,4 @@ def loopn : nat -> Ω1s1 :=
     }
   ]
 
-#normalize {loopn 45}
+#normalize {loopn 5}
```
Comment (favonia) [inline test/circle.cooltt:20]: :laughing:

Comment (jonsterling) [reply test/circle.cooltt:20]: LOL

I wonder who can guess why I had to do this

Comment (favonia) [reply test/circle.cooltt:20]: Well, you seemed to have removed all the optimizations that made 45+ possible...

Comment (jonsterling) [reply test/circle.cooltt:20]: @favonia Yes precisely 😆 I couldn't understand some of them, so I just deleted them 🤦🏼

Comment (jonsterling) [reply test/circle.cooltt:20]: But also, I think not only are the missing optimizations the cause, but the fact that I implemented the reachability check in a very very stupid way.

**Comment by jonsterling** [comment]
I seem to have broken _everything_ LOL. Probably due to some mistake in my incompetent editing of the `kado` library.

**Comment by jonsterling** [comment]
I Upgraded to the new branch! 😎 Right now cooltt thinks that all propositions are true 😝

**Comment by jonsterling** [comment]
looks like you fixed it LOL

File: src/core/Syntax.ml [hunk 2]
```diff
@@ -281,7 +281,7 @@ struct
     | Cof (Cof.Le (r, Dim0)) ->
       Format.fprintf fmt "%a = 0" (pp env P.(left_of cof_le)) r
     | Cof (Cof.Le (r, s)) ->
-      Format.fprintf fmt "%a ≤ %a" (pp env P.(left_of cof_le)) r (pp env P.(right_of cof_le)) s
+      Format.fprintf fmt "%a %a %a" (pp env P.(left_of cof_le)) r Uuseg_string.pp_utf_8 "≤" (pp env P.(right_of cof_le)) s
```
Comment (jonsterling) [inline src/core/Syntax.ml:284]: good catch, thanks!

**Comment by favonia** [comment]
Alright I'm not super motivated to somehow print "i <= j && j <= i" as "i = j". At least not until cooltt has serious formalization.

**Comment by jonsterling** [comment]
I am also not motivated to worry about that...

**Review by favonia — APPROVED** [review]
:genie:

**Pull request merged**